modules/xmpp/JingleSessionPC.js

```diff
@@ -2512,12 +2520,24 @@ export default class JingleSessionPC extends JingleSession {
             );
         const removedAnySSRCs = sdpDiffer.toJingle(remove);
 
+        // context a common object for one run of ssrc update (source-add and source-remove) so we can match them if we
+        // need to
+        const ctx = {};
+
         if (removedAnySSRCs) {
             logger.info(`${this} Sending source-remove`);
             logger.debug(remove.tree());
             this.connection.sendIQ(
-                remove, null,
-                this.newJingleErrorHandler(remove), IQ_TIMEOUT);
+                remove,
+                () => {
+                    this.room.eventEmitter.emit(XMPPEvents.SOURCE_REMOVE, this, ctx);
+                },
+                () => {
+                    this.newJingleErrorHandler(remove);
```

Comment:
newJingleErrorHandler creates a method which needs to be called with an error

Reply:
Fixed.